### ISSUE TITLE
Tab nav link error count is not correctly pluralised for screen readers

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#9521]
### Issue Summary

While investigating https://github.com/wagtail/wagtail/issues/9497 it seems there is a similar issue on the Tab nav link (the tab buttons) that does not correctly pluralise the error counts.

<img width="827" alt="Screen Shot 2022-10-28 at 7 31 03 am" src="https://user-images.githubusercontent.com/1396140/198402156-3aeecf5d-a54c-47a6-9136-2e6910470fd3.png">

### Steps to Reproduce

1. On bakerydemo, edit a page, create a single validation error (e.g. remove the title and make it a blank title)
2. Click save draft, the page will reload with the error count shown in the tab (e.g. `Content (1)`
3. Right-click on the red "1" in a circle on the tab and inspect it in your browser's developer tools
4. Observe that the label for screen readers is "Errors Count: 1 Content"
5. Expected: This label should be correctly labelled for single / plural errors and be easy to read
6. Actual: It uses `errors` not `error` and actually is confusing as it seems like the error is with Content but Content here refers to the tab.

<img width="1513" alt="Screen Shot 2022-10-28 at 7 34 28 am" src="https://user-images.githubusercontent.com/1396140/198402679-c5077c66-6d40-45b8-b767-f6d6e61cc37a.png">

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (no)

### Potential solution

In the file - https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/shared/tabs/tab_nav_link.html

* We should reverse the DOM order so that the `{{ title }}` is before the error count div - this will help with readability and also honours the visual styling where the number is shown after the tab title.
* The `data-tabs-errors-count` could be changed to be the visual representation only (via aria-hidden or similar)
* I would think we should not be using `Count` as a proper noun and instead have a lowercase `c` for `count`. Maybe not even Errors, this is probably just a grammar typo.
* We can then use a different translation mechanism for the error count - see https://docs.djangoproject.com/en/4.1/topics/i18n/translation/#std:templatetag-blocktrans so that the label for screen readers contains the whole thing.

Something like the following could be a good start, but this will read "Content 1 error" and "Content 3 errors" - we should be careful with assuming the value of the tab and trying to put that in the sentence. Maybe an easy way around this is to wrap the error label in brackets instead.

```
<a id="tab-label-{{ tab_id }}" href="#tab-{{ tab_id }}" class="w-tabs__tab {{ classes }}" role="tab" aria-selected="false" tabindex="-1">
    {{ title }}
    <div data-tabs-errors class="w-tabs__errors {% if errors_count %}!w-flex{% endif %}">
        <span data-tabs-errors-count aria-hidden="true">{{ errors_count }}</span>
        <span class="w-sr-only">{% blocktrans trimmed count errors_count=errors_count %}1 error{% plural %}{{ errors_count }} errors{% endblocktranslate %}</span>
    </div>
</a>
```

### Technical details

-   Python version: 3.0
-   Django version: 4.1
-   Wagtail version: 4.2a (in development)
-   Browser version: Firefox 106

**Comment by Headon1729** [comment]
Hey @lb- 
I would like to work on this issue if you are not working on it

**Comment by lb-** [comment]
@Headon1729 go for it - please start with reproducing the issue in a different browser and confirming you can see the problem. Maybe add another screenshot. 

That way we can remove the unconfirmed label, plus that will help you understand the issue better.

**Comment by Buduzz** [comment]
Hello @lb- can I work on this, attached are test images from different browsers  
![image](https://user-images.githubusercontent.com/115478779/198698801-8989783e-a4c3-470d-8299-fbdd2688092f.png)
![image](https://user-images.githubusercontent.com/115478779/198699174-9f773da0-49aa-46ce-832c-87eb9f471058.png)
For the error count label can the error message be: `Found {num} error(s) in content tab`?

**Comment by Headon1729** [comment]
Hey @Buduzz
I'm working on this issue & would soon make the PR. Can you work on any other issue

**Comment by salty-ivy** [comment]
@Headon1729  are you still working on it? shall i take this one up?

**Comment by salty-ivy** [comment]
@lb- should pick this one up seems like @Headon1729  is not responding , off the issue

**Comment by lb-** [comment]
Go for it @salty-ivy

**Comment by salty-ivy** [comment]
```
<a id="tab-label-{{ tab_id }}" href="#tab-{{ tab_id }}" class="w-tabs__tab {{ classes }}" role="tab" aria-selected="false" tabindex="-1">
    {{ title }}
    <div data-tabs-errors class="w-tabs__errors {% if errors_count %}!w-flex{% endif %}">
        <span data-tabs-errors-count aria-hidden="true">{{ errors_count }}</span>
        <span class="w-sr-only">{% blocktrans trimmed count errors_count=errors_count %}1 error{% plural %}{{ errors_count }} errors{% endblocktranslate %}</span>
    </div>
</a>
```
this seems to throw an error, i think we need to find the other way around using trans and if else logic

![image](https://user-images.githubusercontent.com/74553951/202287772-a8fa1903-e42e-4ad4-b400-9eb47fb17f61.png)

**Comment by lb-** [comment]
Sorry that code was just a rough start/guide - I didn't actually test it.

**Comment by salty-ivy** [comment]
I even tried this logic , given the previous error and fact that only else statement is invoked make it seems like errors_count is not actually a number or int

```
<a id="tab-label-{{ tab_id }}" href="#tab-{{ tab_id }}" class="w-tabs__tab {{ classes }}" role="tab" aria-selected="false" tabindex="-1">
    {{ title }}
    <div data-tabs-errors class="w-tabs__errors {% if errors_count %}!w-flex{% endif %}">
        {% if errors_count > 1 %}
            <span data-tabs-errors-count>{{ errors_count }}</span>
            <span class="w-sr-only">{% trans "errors" %}</span>
        {% else %}
            <span data-tabs-errors-count>{{ errors_count }}</span>
            <span class="w-sr-only">{% trans "error" %}</span>
        {% endif %}
    </div>
</a>
```

**Comment by lb-** [comment]
@salty-ivy ok - do a bit more digging - feel free to propose a better solution. My suggestion above for a fix was only done super quickly without spending too much time on this.

**Comment by salty-ivy** [comment]
> If you search through the code for `tab_nav_link` you will find it is used in lots of places via `{% include 'wagtailadmin/shared/tabs/tab_nav_link.html'`
> 
> But the main one for this issue is when it's used as part of the tabbed interface.
> 
> https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/panels/tabbed_interface.html
> 
> Which is part of this code https://github.com/wagtail/wagtail/blob/main/wagtail/admin/panels.py#L587
> 
> Which uses `PanelGroup` & `BoundPanel` from that file.
> 
> It looks like the errors come from the `get_context_data` method so maybe start your logging there.
> 
> https://github.com/wagtail/wagtail/blob/main/wagtail/admin/panels.py#L801

yea as errors_count is not set correctly that's why i was looking for context code 😅

**Comment by salty-ivy** [comment]
@lb- are we sure about the fact that context has been passed through normal way through views and classes or is there any unknown way like may be through static js or any other way ?

**Comment by lb-** [comment]
Sorry - not sure on that.

**Comment by salty-ivy** [comment]
@lb-  so far i have looked into it a little couldn't find any reference to errors_count, 
so far i have found this react tsx  file that refers to errors_count, https://github.com/wagtail/wagtail/blob/119f288a3cf91775687aafffe19ac4c92bb45e0a/wagtail/admin/templates/wagtailadmin/shared/tabs/tab_nav_link.stories.tsx#L28

i couldn't figure out much about it, could you please have a look,

**Comment by salty-ivy** [comment]
would soon still look more into it

**Comment by salty-ivy** [comment]
> even though i am not using the errors_count in the template it still renders out, i believe it might have been injected from js, or any other way ![image](https://user-images.githubusercontent.com/74553951/202493035-940b60d6-fd29-4e6d-ba55-0dad8cca71a1.png) here i have removed the errors_count
> 
> ![image](https://user-images.githubusercontent.com/74553951/202493479-8033b685-7266-4b9f-9cb8-85ae9dd8b981.png) and here you can see it still renders out
> 
> ![image](https://user-images.githubusercontent.com/74553951/202493671-6e77fde1-24c3-4c52-b252-d5cf9cd46555.png)

also the div still inherits ```!w-flex``` class which was not supposed to be assigned to it considering the code changes

**Comment by lb-** [comment]
@salty-ivy  I think you are right about the JS stuff - searching in the code for `data-dabts-errors` uncovers this file.

client/src/entrypoints/admin/page-editor.js see `initErrorDetection`

**Comment by salty-ivy** [comment]
Would look into it soon

**Comment by salty-ivy** [comment]
errors_count variable in the template is not making any difference it contains nothing all the class assignment and error text in the template is being added by JS  ```initErrorDetection```

**Comment by salty-ivy** [comment]
@lb-  i have made the PR, kindly review at your ease and let me know about any suggestions and ideas, thank you
